predict/src/main.rs

```diff
@@ -110,6 +117,7 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
                 post_filters.iter().for_each(|filter| filter.filter(&mut s));
                 if args.predict_tags {
                     s.fill_tags();
+                    pattern_match_tagger.filter(&mut s);
```

Comment:
Even if pattern_match_tagger has no entry, filter() is performed and the entire sentence is scanned.
This scan is redundant, and it would be better to implement pattern_match_tagger with Option.